revolutionpi/digital_interrupts.go

```diff
@@ -0,0 +1,107 @@
+//go:build linux
+
+// Package revolutionpi implements the Revolution Pi board GPIO pins.
+package revolutionpi
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"go.viam.com/rdk/components/board"
+)
+
+const (
+	inputModeOffset = 88
+)
+
+type digitalInterrupt struct {
+	PinName          string // Variable name
+	Address          uint16 // Address of the byte in the process image
+	Length           uint16 // length of the variable in bits. Possible values are 1, 8, 16 and 32
+	BitPosition      uint8  // 0-7 bit position, >= 8 whole byte, only used if the digital input pin is given
+	ControlChip      *gpioChip
+	outputOffset     uint16
+	inputOffset      uint16
+	enabled          bool
+	interruptAddress uint16
+}
+
+func (di *digitalInterrupt) initialize() error {
+	var addressInputMode uint16
+
+	// read from the input mode byte to determine if the pin is configured for counter/interrupt mode
+	// determine which address to check for the input mode based on which pin was given in the request
+	switch {
+	case di.isInputCounter():
+		addressInputMode = (di.Address - di.inputOffset - inputWordToCounterOffset) >> 2
+
+		// record the address for the interrupt
+		di.interruptAddress = di.Address
+	case di.isDigitalInput():
+		// depending on whether the request came from the first or second set of digital input pins, add 0 or 8
+		firstOrSecondHalf := (di.Address - di.inputOffset) << 3
+		addressInputMode = firstOrSecondHalf + uint16(di.BitPosition)
+		di.interruptAddress = di.inputOffset + inputWordToCounterOffset + addressInputMode*4
+	default:
+		return errors.New("pin is not a digital input pin")
+	}
```

Comment:
more aliasing fun. this time I stored the address on the interrupt struct to reduce the number of times this logic needed to happen